frontend/src/e2e-playwright/specs/5_employee/employee-pin.spec.ts

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { newBrowserContext } from 'e2e-playwright/browser'
+import EmployeeNav from 'e2e-playwright/pages/employee/employee-nav'
+import { waitUntilEqual } from 'e2e-playwright/utils'
+import config from 'e2e-test-common/config'
+import { resetDatabase } from 'e2e-test-common/dev-api'
+import { Page } from 'playwright'
+import { EmployeePinPage } from '../../pages/employee/employee-pin'
+import { Fixture } from '../../../e2e-test-common/dev-api/fixtures'
+
+let page: Page
+let nav: EmployeeNav
+let pinPage: EmployeePinPage
+
+beforeEach(async () => {
+  await resetDatabase()
+
+  page = await (await newBrowserContext()).newPage()
+  await page.goto(config.employeeUrl)
+  nav = new EmployeeNav(page)
+  await nav.login('ADMIN')
+  pinPage = new EmployeePinPage(page)
+})
+
+describe('Employees PIN', () => {
+  beforeEach(async () => {
+    await nav.openAndClickDropdownMenuItem('pinCode')
+  })
+
+  test('shows a warning if PIN is too easy, and warning disappears once PIN is valid', async () => {
```

Comment:
This isn't exactly what was changed here but still: what should happen when the user inputs four letters or other non-numeric symbols? Currently it looks like the input only checks that input length == 4:

```ts
  function isValidNumber(pin: string) {
    const int = parseInt(pin)
    const str = int.toString()
    return str.length === 4
  }
```

-> `0x1000` would be accepted as `parseInt` (as no `radix` is defined) would interpret is as hexadecimal 4096 -> 4 characters as a string. Currently the PIN would still be stored as the string "0x1000" but that would be confusing when prompted for a 4-number token.

Also, looks like anything beginning with 0 wouldn't be a valid PIN. Is that desired?